website/source/docs/internals/anti-entropy.html.markdown

```diff
@@ -0,0 +1,131 @@
+configuration options available. This is because the agent is responsible for
+generating information about its services and their health through the use of
+[health checks](/docs/agent/checks.html).
+
+<a name="catalog"></a>
+#### Catalog
+
+Consul's service discovery is backed by a service catalog. This catalog is
+formed by aggregating information submitted by the agents. The catalog maintains
+the high-level view of the cluster, including which services are available,
+which nodes run those services, health information, and more. The catalog is
+used to expose this information via the various interfaces Consul provides,
+including DNS and HTTP.
+
+Services and checks within the context of the catalog have a much more limited
+set of fields when compared with the agent. This is because the catalog is only
+responsible for recording and returning information *about* services, nodes, and
+health.
+
+The catalog is maintained only by server nodes. This is because the catalog is
+replicated via the [Raft log](/docs/internals/consensus.html) to provide a
+consolidated and consistent view of the cluster.
+
+<a name="anti-entropy"></a>
+### Anti-Entropy
+
+Now that we have covered the functions of the agent and the catalog, we
+need a mechanism to perform synchronization between the two. In Consul, this
+method is known as anti-entropy.
+
+Anti-entropy is a syncronization of the local agent state and the catalog. For
+example, when a user registers a new service or check with the agent, the agent
+in turn notifies the catalog that this new check exists. Similarly, when a check
+is deleted from the agent, it is consequently removed from the catalog as well.
+
+Anti-entropy is also used to update availability information. As agents run
+their health checks, their status may change, in which case their new status
+is synced to the catalog. Using this information, the catalog can respond
+intelligently to queries about its nodes and services based on their
+availability.
+
+During this synchronization, the catalog is also checked for correctness. If
+any services or checks exist in the catalog that the agent is not aware of, they
+will be automatically removed to make the catalog reflect the proper set of
+services and health information for that agent.
+
+### Periodic Synchronization
+
+In addition to running when changes to the agent occur, anti-entropy is also a
+long-running process which periodically wakes up to sync service and check
+status to the catalog. This ensures that the catalog closely matches the agent's
+true state.
+
+The amount of time between periodic anti-entropy runs will vary based on cluster
+size to avoid saturation. The table below describes the periodic sync times and
+how they change as the Consul cluster grows.
+
+<table class="table table-bordered table-striped">
+  <tr>
+    <th>Cluster Size</th>
+    <th>Periodic Sync Interval</th>
+  </tr>
+  <tr>
+    <td>1 - 128</td>
+    <td>15 seconds</td>
```

Comment:
I think these times seem very aggressive. I think the default is 60 seconds for the smallest of clusters?